src/io/flutter/utils/CustomIconMaker.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2017 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.utils;
+
+import com.intellij.ui.ColorUtil;
+import com.intellij.ui.LayeredIcon;
+import com.intellij.util.ui.GraphicsUtil;
+import com.intellij.util.ui.UIUtil;
+import icons.FlutterIcons;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.geom.Rectangle2D;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CustomIconMaker {
+  private static final Color normalColor = ColorUtil.fromHex("231F20");
+
+  private final Map<String, Icon> iconCache = new HashMap<>();
+
+  public CustomIconMaker() {
+
+  }
```

Comment:
Nit: why the explicit default constructor?

Reply:
Hmm, habit? Should we avoid?

Reply:
Nope.  Just curious!